eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -542,7 +542,7 @@ runPLCScript proxy lang (CostModel cost) scriptbytestring units ds =
     scriptbytestring
     ds of
     (_, Left e) -> explainPlutusFailure proxy lang scriptbytestring e ds (CostModel cost) units
```

Comment:
How is this actually going to look in the events? From the look of this function, here we're going to get a predicate failure (?) with the _serialized_ version of `PlutusDebug`, whereas in the success case we'll just get the structure itself? It seems like that might be a bit of a pain to analyze consistently.

Reply:
This is all related to https://github.com/input-output-hk/cardano-ledger/pull/2670/files#r816904931

`evalScripts` is used both in the case that `IsValid` is true and false. Currently, in this PR, I've avoided the `IsValid = False` case (which is maybe a mistake) with respect to the events. This left here is currently only feeding into a predicate failure, not the events.